keps/sig-instrumentation/1661-event-api-migration/README.md

```diff
@@ -0,0 +1,422 @@
+
+Most sections and design details are copied from the original design doc: [Make Kubernetes Events Useful and Safe](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/instrumentation/events-redesign.md).
+
+The new Event API v1beta1 has been implemented for a while and currently only scheduler was migrated to use it. This KEP proposes to migrate all the remaining Kubernetes components to use the new Event API.
+
+## Motivation
+
+There's a relatively wide agreement that current implementation of Events in Kubernetes is problematic. Events are supposed to give app developer insight into what's happening with his/her app. Important requirement for Event library is that it shouldn't cause/worsen performance problems in the cluster.
+
+The problem is that neither of those requirements are actually met. Currently Events are extremely spammy (e.g. Event is emitted when Pod is unable to schedule every few seconds) with unclear semantics (e.g. Reason was understood by developers as "reason for taking action" or "reason for emitting event"). Also there are well known performance problems caused by Events (e.g. [#47366](https://github.com/kubernetes/kubernetes/issues/47366), [#47899](https://github.com/kubernetes/kubernetes/issues/47899)) - Events can overload API server if there's something wrong with the cluster (e.g. some correlated crashloop on many Nodes, user created way more Pods than fit on the cluster which fail to schedule repeatedly). This was raised by the community on number of occasions.
+
+### Goals
+
+* Update Event semantics such that they'll be considered useful by app developers.
+* Reduce impact that Events have on the system's performance and stability.
+
+### Non-Goals
+
+* Persist Events outside of etcd or for a longer time.
+
+## Proposal
+
+The new Event API makes all semantic information about events first-class fields, allowing better deduplication and querying. The API changes aim to:
+* make it easy to list all interesting Events in common scenarios using kubectl:
+  * Listing Events mentioning given Pod,
+  * Listing Events emitted by a given component (e.g. Kubelet on a given machine, NodeController),
+* make timestamps precise enough to allow better events correlation,
+* update the field names to better indicate their function.
+
+In addtion, we will improve 'Event series' detection and send only 'series start' and 'series finish' Events, and add more aggressive backoff policy for Events.
+
+After the migration is done, users and administrators:
+* will be able to better track interesting changes in the state of objects they're interested in,
+* will be convinced that Events do not destabilize.
+
+## Design Details
+
+Note that the design details talk about all aspects of the whole Event redesign effort, including API changes where the implementation has been done in API group v1beta1.
+
```

Comment:
Usability issues section is actually pretty useful and really important one:
https://github.com/kubernetes/community/blob/master/contributors/design-proposals/instrumentation/events-redesign.md#usability-issues-with-the-current-api

Should be copied here too